.github/workflows/build-decider.yml

```diff
@@ -0,0 +1,80 @@
+name: Decide jobs
+
+# Set BUILD_TARGETS either here or as an environment variable in repository settings
+# using the template given below, omitting the build targets you don't need. If BUILD_TARGETS
+# is not set or empty, no filtering is applied and all targets will be built.
+#
+# For example, the old FAST_MODE would be
+#   BUILD_TARGETS: ^(arm-linux|linux64)$
+#
+# A template for all currently supported targets is
+#   BUILD_TARGETS: ^(arm-linux|linux64|linux64_fuzz|linux64_multiprocess|linux64_nowallet|linux64_sqlite|linux64_tsan|linux64_ubsan|mac|win64)$
+
+env:
+  BUILD_TARGETS: ${{ vars.BUILD_TARGETS }}
+
+on:
+  workflow_call:
+    outputs:
+      run-arm-linux:
+        value: ${{ jobs.decide-jobs.outputs.run-arm-linux }}
+      run-linux64:
+        value: ${{ jobs.decide-jobs.outputs.run-linux64 }}
+      run-linux64_fuzz:
+        value: ${{ jobs.decide-jobs.outputs.run-linux64_fuzz }}
+      run-linux64_multiprocess:
+        value: ${{ jobs.decide-jobs.outputs.run-linux64_multiprocess }}
+      run-linux64_nowallet:
+        value: ${{ jobs.decide-jobs.outputs.run-linux64_nowallet }}
+      run-linux64_sqlite:
+        value: ${{ jobs.decide-jobs.outputs.run-linux64_sqlite }}
+      run-linux64_tsan:
+        value: ${{ jobs.decide-jobs.outputs.run-linux64_tsan }}
+      run-linux64_ubsan:
+        value: ${{ jobs.decide-jobs.outputs.run-linux64_ubsan }}
+      run-mac:
+        value: ${{ jobs.decide-jobs.outputs.run-mac }}
+      run-win64:
+        value: ${{ jobs.decide-jobs.outputs.run-win64 }}
+
+jobs:
+  decide-jobs:
+    name: Decide jobs
+    runs-on: ubuntu-24.04
+    outputs:
+      run-arm-linux: ${{ steps.script.outputs.run-arm-linux }}
+      run-linux64:  ${{ steps.script.outputs.run-linux64 }}
+      run-linux64_fuzz: ${{ steps.script.outputs.run-linux64_fuzz }}
+      run-linux64_multiprocess: ${{ steps.script.outputs.run-linux64_multiprocess }}
+      run-linux64_nowallet: ${{ steps.script.outputs.run-linux64_nowallet }}
+      run-linux64_sqlite: ${{ steps.script.outputs.run-linux64_sqlite }}
+      run-linux64_tsan: ${{ steps.script.outputs.run-linux64_tsan }}
+      run-linux64_ubsan: ${{ steps.script.outputs.run-linux64_ubsan }}
+      run-mac: ${{ steps.script.outputs.run-mac }}
+      run-win64: ${{ steps.script.outputs.run-win64 }}
```

Comment:
UGH; I kinda hate the idea of needing to change 5 different spots when we add a new host or job :/

Reply:
I kinda feel this isn't useful? And just complicates actions more? I don't see a use in only running 1 job in GitHub Actions; as @UdjinM6, these runners are free and fast enough. And if you only want certain jobs, just cancel the others in the UI?

Reply:
The repeated enumeration is because GitHub doesn't provide a way to "export all outputs", adding new jobs isn't very frequent and I did find narrowing jobs to be valuable when needing quick feedback on whether changes to workflows, especially in the latter half, behave properly before running them on all variants.

Reply:
Why not just cancel jobs?

Reply:
Because GitHub Actions doesn't support arbitrary job cancellation ([source](https://stackoverflow.com/a/61832535))

You can do restarts

![Restart icon](https://github.com/user-attachments/assets/e0d01fb7-92e8-4c61-84c7-22fb8f256917)

Not so much cancellations

![No cancel icon](https://github.com/user-attachments/assets/604e136e-e3a2-4841-a8d8-d44a02414b5e)

Reply:
I was a bit concerned about this too initially. Would be much easier if GH `if` syntax would support regex but it doesn't 🤷‍♂️ I tried implementing alternative workarounds but couldn't get them working... On a positive side, the set of build targets isn't going to change that often. So overall, it's ok-ish to have a not-super-elegant solution in this case imo.